ticker/src/main/java/com/robinhood/ticker/LevenshteinUtils.java

```diff
@@ -60,38 +61,42 @@
         }
 
         // Reverse trace the matrix to compute the necessary actions
-        int i = numRows - 1;
-        int j = numCols - 1;
-        int resultIndex = resultLength - 1;
-        while (resultIndex >= 0) {
-            if (i == 0) {
+        final List<Integer> resultList = new ArrayList<>(resultLength * 2);
+        int row = numRows - 1;
+        int col = numCols - 1;
+        while (row > 0 || col > 0) {
+            if (row == 0) {
                 // At the top row, can only move left, meaning insert column
-                result[resultIndex] = ACTION_INSERT;
-                j--;
-            } else if (j == 0) {
+                resultList.add(ACTION_INSERT);
+                col--;
+            } else if (col == 0) {
                 // At the left column, can only move up, meaning delete column
-                result[resultIndex] = ACTION_DELETE;
-                i--;
+                resultList.add(ACTION_DELETE);
+                row--;
             } else {
-                final int top = matrix[i-1][j];
-                final int left = matrix[i][j-1];
-                final int topLeft = matrix[i-1][j-1];
+                final int insert = matrix[row][col-1];
+                final int delete = matrix[row-1][col];
+                final int replace = matrix[row-1][col-1];
 
-                if (topLeft <= top && topLeft <= left) {
-                    result[resultIndex] = ACTION_SAME;
-                    i--;
-                    j--;
-                } else if (top <= left) {
-                    result[resultIndex] = ACTION_DELETE;
-                    i--;
+                if (insert < delete && insert < replace) {
+                    resultList.add(ACTION_INSERT);
+                    col--;
+                } else if (delete < replace) {
+                    resultList.add(ACTION_DELETE);
+                    row--;
                 } else {
-                    result[resultIndex] = ACTION_INSERT;
-                    j--;
+                    resultList.add(ACTION_SAME);
+                    row--;
+                    col--;
                 }
             }
-            resultIndex--;
         }
 
+        final int resultSize = resultList.size();
+        final int[] result = new int[resultSize];
+        for (int i = 0; i < resultSize; i++) {
+            result[resultSize - 1 - i] = resultList.get(i);
+        }
         return result;
```

Comment:
uh hm just realized that the above can't return empty list because it needs to return the number of columns with no modifications, and this iteration also adds the updates in reverse since we do reverse trace of the matrix to get the modifications. eh i'll just leave it for now lol.